src/service/viewport-impl.js

```diff
@@ -342,6 +348,18 @@ export class Viewport {
   }
 
   /**
+   * Returns the first viewport rect.
+   * @return {!../layout-rect.LayoutRectDef}}
+   */
+  getFirstViewportRect() {
+    if (this.firstViewportRect_ == null) {
```

Comment:
this.firstViewportRect_ value is cached. what if in the case of viewport change?